Applications/Spire/Source/Ui/ColorSwatch.cpp

```diff
@@ -0,0 +1,87 @@
+#include "Spire/Ui/ColorSwatch.hpp"
+#include "Spire/Spire/Dimensions.hpp"
+#include "Spire/Spire/LocalValueModel.hpp"
+#include "Spire/Styles/ChainExpression.hpp"
+#include "Spire/Styles/LinearExpression.hpp"
+#include "Spire/Styles/RevertExpression.hpp"
+#include "Spire/Styles/TimeoutExpression.hpp"
+#include "Spire/Ui/Box.hpp"
+#include "Spire/Ui/Layouts.hpp"
+
+using namespace boost::posix_time;
+using namespace Spire;
+using namespace Spire::Styles;
+
+namespace {
+  auto DEFAULT_STYLE() {
+    auto style = StyleSheet();
+    style.get(Any()).
+      set(border(scale_width(1), QColor(Qt::transparent))).
+      set(horizontal_padding(scale_width(1))).
+      set(vertical_padding(scale_height(1)));
+    style.get(Hover() || Focus()).
+      set(border_color(QColor(0x4B23A0)));
+    style.get(Drag()).
+      set(BackgroundColor(QColor(255, 255, 255, 204))).
+      set(border_color(QColor(0x4B23A0)));
+    style.get(Any() >> Body()).
```

Comment:
Why is the descendant selector used here? Could you instead use either the top selector or select the component directly?

Reply:
Initially, I used the top selector, which worked well for ColorSwatch alone. However, when the ColorSwatch is used within the ColorPicker, it seems that styles are applied to the entire ColorSwatch instead of its Body().
I can update code to apply these styles to the component directly.

Reply:
I will replace the Body selector with the Swatch selector to resolve this issue.